google-cloud-pubsub/lib/google/cloud/pubsub/subscriber/enumerator_queue.rb

```diff
@@ -30,9 +30,11 @@ def push obj
             @queue.push obj
           end
 
-          def dump_queue
+          def quit_and_dump_queue
```

Comment:
It seems to me that adding a separate method that just pushes the sentinel might be easier to reason about than this addition of a side-effect to `dump_queue`, which was previously a simple operation.

Reply:
I'd rather have these operations happen simultaneously. This is only to be used as part of the recovery for a failed stream, which is why I felt a longer method name was warranted.